src/plugins/formulas/formulas.js

```diff
@@ -384,7 +398,7 @@ export class Formulas extends BasePlugin {
   isFormulaCellType(row, column, sheet = this.sheetId) {
     const cellType = this.getCellType(row, column, sheet);
 
-    return cellType === 'FORMULA' || cellType === 'MATRIX';
+    return cellType === 'FORMULA' || cellType === 'ARRAYFORMULA';
   }
```

Comment:
It does not make sense to fix this method each time something changes in HF and HF has exactly the same method already: https://github.com/handsontable/hyperformula/blob/b303691371d5ab9a66f5eae050cdfbf596530613/src/HyperFormula.ts#L3096 can we just use HF instead?

```ts
isFormulaCellType(row, column, sheet = this.sheetId) {
  return this.engine.doesCellHaveFormula({ row, column, sheet });
}
```

Reply:
+1, currently `doesCellHaveFormula` has the exact same condition built-in: https://github.com/handsontable/hyperformula/blob/072f829bff49a389b2b4be8649cf654bdb998cf2/src/HyperFormula.ts#L3096 (in `<1.0.0` the right-hand side of the condition was missing there)

Reply:
https://github.com/handsontable/handsontable/pull/8669/commits/03704761f213f093b0c63e46d5b9297c562dace5